src/EnergyPlus/EvaporativeCoolers.cc

```diff
@@ -4085,21 +4110,21 @@ namespace EvaporativeCoolers {
         // na
 
         // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
-        static bool MyOneTimeFlag(true); // one time flag
         static Array1D_bool MyEnvrnFlag;
         static Array1D_bool MyFanFlag;
         static Array1D_bool MyZoneEqFlag; // used to set up zone equipment availability managers
         int Loop;
         static bool ZoneEquipmentListChecked(false); // True after the Zone Equipment List has been checked for items
         Real64 TimeElapsed;
 
-        if (MyOneTimeFlag) {
+        if (InitZoneHVACEvapCoolerOneTimeFlag) {
             MySizeFlag.dimension(NumZoneEvapUnits, true);
             MyEnvrnFlag.dimension(NumZoneEvapUnits, true);
             MyFanFlag.dimension(NumZoneEvapUnits, true);
             MyZoneEqFlag.allocate(NumZoneEvapUnits);
             MyZoneEqFlag = true;
-            MyOneTimeFlag = false;
+            InitZoneHVACEvapCoolerOneTimeFlag = false;
+            ZoneEquipmentListChecked = false;
```

Comment:
I know you didn't add any arrays here, but this is another place where the Evap structure could just have booleans:

```
struct Evap {
  bool mySize = true;
  bool myEnvrn = true;
  bool myFan = true;
  bool myZoneEq = true;
}
```

Then we don't have a bunch of extra static arrays to manage.  Nothing for you to necessarily do here just saying it is a welcomed change whenever you are working on code like this.

Reply:
@Myoldmopar updated Evap structure.